feat!: Add per-device ACB inventory, sleep control, and devtools (#428)
pyenphase/pyenphase: docs/data_ensemble.md

- Combined Encharge + ACB SOC/capacity is available in [EnvoyData.battery_aggregate](#pyenphase.EnvoyData.battery_aggregate), modeled by [EnvoyBatteryAggregate](#pyenphase.models.acb.EnvoyBatteryAggregate).
- Per-device ACB data is available in [EnvoyData.acb_inventory](#pyenphase.EnvoyData.acb_inventory), keyed by serial number and modeled by [EnvoyACB](#pyenphase.models.acb.EnvoyACB).
- The number of ACB batteries reported in production storage can be read from [Envoy.acb_count](#pyenphase.Envoy.acb_count).

Per-device ACB fields include state and sensor values such as `sleep_enabled`, `sleep_state`, `sleep_min_soc`, `sleep_max_soc`, `percent_full`, `charge_status`, `communicating`, `operating`, `producing`, `last_report_watts`, `max_report_watts`, and `last_report_date`.

```python
print(f"ACB count: {envoy.acb_count}")

if envoy.data.acb_inventory:
    for serial, acb in envoy.data.acb_inventory.items():
        print(serial, acb.sleep_state, acb.percent_full, acb.last_report_watts)
```

ACB sleep control is available with [Envoy.set_acb_sleep](#pyenphase.Envoy.set_acb_sleep) and [Envoy.clear_acb_sleep](#pyenphase.Envoy.clear_acb_sleep).

```python

> await envoy.set_acb_sleep(

Reviewer: Does this example mean that the battery will go into sleep mode (no charge nor discharge) when soc reaches this interval?

Author: Not exactly - the battery will charge or discharge *to reach* the target SoC before entering sleep mode, not when it happens to be within the range. Specifically: if the battery is above `sleep_max_soc` it will discharge down to that level; if it's below `sleep_min_soc` it will charge up to that level; then it sleeps. The values in the example (`10`/`20`) are just illustrative numbers. I'll add a clarifying note to the docs explaining this behavior.

```python
status = await envoy.go_off_grid()
if status["mains_admin_state"] != "open":
    raise RuntimeError("Failed to go off-grid")

status = await envoy.go_on_grid()
if status["mains_admin_state"] != "closed":
    raise RuntimeError("Failed to go on-grid")
```

[open_dry_contact](#pyenphase.Envoy.open_dry_contact) and [close_dry_contact](#pyenphase.Envoy.close_dry_contact) can be used to control dry contacts.

```python
status = await envoy.close_dry_contact(id)
print(envoy.data.dry_contact_status[id].status)

status = await envoy.open_dry_contact(id)
print(envoy.data.dry_contact_status[id].status)
```

Dry contact information is available in [EnvoyData.dry_contact_status](#pyenphase.EnvoyData.dry_contact_status) and [EnvoyData.dry_contact_settings](#pyenphase.EnvoyData.dry_contact_settings).

[Envoy.update_dry_contact](#pyenphase.Envoy.update_dry_contact) can be used to update settings. Use with care and only if fully aware of the impact.

```python
from typing import Any

new_setting: dict[str, Any] = {
        "id": id,
        "load_name": load_name,
}

status = await envoy.update_dry_contact(new_setting)
print(status)
```

## Enphase AC Battery (ACB) data

> The ACB integration exposes both aggregate and per-device battery data:

Reviewer: Integration is an HOme Assistant term. Here we are still more generic

```suggestion
Both ACB aggregate and per-device battery data are exposed:
```
